src/tests/ftest/util/nvme_utils.py

```diff
@@ -0,0 +1,425 @@
+            tmp_dict = {}
+            for daos_io_server in range(len(self.daos_io_servers)):
+                tmp_disk_list = []
+                for disk in (self.server_managers[0].manager.job.yaml.
+                             server_params[daos_io_server].bdev_list.value):
+                    if disk in nvme_readlink[server].keys():
+                        size = int(nvme_lsblk[server]
+                                   [nvme_readlink[server][disk]])
+                        tmp_disk_list.append(size)
+                    else:
+                        self.fail("Disk {} can not found on server {}"
+                                  .format(disk, server))
+                tmp_dict[daos_io_server] = tmp_disk_list
+            drive_info[server] = tmp_dict
+
+        return self.get_max_capacity(drive_info)
+
+    def start_ior_thread(self, results, operation='Write'):
+        """Start IOR write/read threads and wait until all threads are finished.
+
+        Args:
+            results (queue): queue for returning thread results
+            operation (str): IOR operation for read/write on same container.
+
+        Returns:
+            None
+        """
+        mpio_util = MpioUtils()
+        if mpio_util.mpich_installed(self.hostlist_clients) is False:
+            self.fail("Exiting Test: Mpich not installed")
+
+        # Define the arguments for the ior_runner_thread method
+        ior_cmd = IorCommand()
+        ior_cmd.get_params(self)
+        ior_cmd.set_daos_params(self.server_group, self.pool)
+        ior_cmd.daos_oclass.update(self.ior_daos_oclass)
+        ior_cmd.api.update(self.ior_apis)
+        ior_cmd.transfer_size.update(self.ior_transfer_size)
+
+        #If IOR write calculate the block size based on server % to fill up
+        if 'Write' in operation:
+            block_size = self.calculate_ior_block_size()
+            ior_cmd.block_size.update('{}'.format(block_size))
+            ior_cmd.flags.update(self.ior_flags)
+            #Store the container UUID and block size used for
+            #reading and verification purpose in future.
+            self.container_info["{}{}{}"
+                                .format(self.ior_daos_oclass,
+                                        self.ior_apis,
+                                        self.ior_transfer_size)] = [str(
+                                            uuid.uuid4()), block_size]
+        elif 'Read' in operation:
+            ior_cmd.flags.update(self.ior_read_flags)
+            #Retrieve the container UUID and block size for reading purpose
+            ior_cmd.block_size.update('{}'
+                                      .format(self.container_info
+                                              ["{}{}{}".format(
+                                                  self.ior_daos_oclass,
+                                                  self.ior_apis,
+                                                  self.ior_transfer_size)][1]))
```

Comment:
Can we do both write/read operation at the same time? Looks like we can do either Write/Read now.

Reply:
Write can actually do both based on self.ior_flags but if user want to do only read can be done based on container info store in the dict. Let me change the name to avoid confusion